app/ui-message/client/ActionButtonSyncer.ts

```diff
@@ -1,29 +1,82 @@
 import { Meteor } from 'meteor/meteor';
-import { IUIActionButton, UIActionButtonContext } from '@rocket.chat/apps-engine/definition/ui';
+import { IUIActionButton, MessageActionContext, TemporaryRoomTypeFilter, UIActionButtonContext } from '@rocket.chat/apps-engine/definition/ui';
 
 import { APIClient } from '../../utils/client';
 import * as TabBar from './actionButtons/tabbar';
+import * as MessageAction from './actionButtons/messageAction';
+import * as MessageBox from './actionButtons/messageBox';
 
-let registeredButtons: Array<IUIActionButton>;
+let registeredButtons: Array<IUIActionButton> = [];
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+const mockButtons = () => ([
+	{
+		context: UIActionButtonContext.MESSAGE_ACTION,
+		icon: 'tab',
+		actionId: 'messageaction',
+		appId: 'mock-app-1',
+		nameI18n: 'test-i18n',
+		when: {
+			messageActionContext: [MessageActionContext.MESSAGE],
+			roomTypes: [TemporaryRoomTypeFilter.PUBLIC_CHANNEL, TemporaryRoomTypeFilter.PRIVATE_CHANNEL],
+			hasOnePermission: ['view-all-teams'],
+			hasAllRoles: ['admin', 'moderator'],
+		},
+	},
+	{
+		context: UIActionButtonContext.MESSAGE_BOX_ACTION,
+		icon: 'tab',
+		actionId: 'messagebox',
+		appId: 'mock-app-1',
+		nameI18n: 'test-i18n',
+		when: {
+			roomTypes: [TemporaryRoomTypeFilter.DIRECT, TemporaryRoomTypeFilter.DIRECT_MULTIPLE],
+			hasAllPermissions: ['create-team', 'create-invite-links'],
+			hasOneRole: ['user', 'admin'],
+		},
+	},
+	{
+		context: UIActionButtonContext.ROOM_ACTION,
+		icon: 'tab',
+		actionId: 'room',
+		appId: 'mock-app-3',
+		nameI18n: 'test-i18n',
+		when: {
+			roomTypes: [TemporaryRoomTypeFilter.PUBLIC_TEAM, TemporaryRoomTypeFilter.PRIVATE_TEAM, TemporaryRoomTypeFilter.LIVE_CHAT],
+			hasOnePermission: ['view-all-teams'],
+			hasAllRoles: ['admin'],
+		},
+	},
+// eslint-disable-next-line @typescript-eslint/no-use-before-define
+] as Array<IUIActionButton>).map(addButton);
```

Comment:
I'm curious, this type coercion cannot be used as return type?

Reply:
That block was a mock for testing the buttons. Removed it :)